File: studio/app/src/androidTest/java/nl/sogeti/android/gpstracker/ng/tracks/TrackListFragmentTest.java
package nl.sogeti.android.gpstracker.ng.tracks;

import android.net.Uri;
import android.support.test.runner.AndroidJUnit4;

import junit.framework.Assert;

import org.jetbrains.annotations.NotNull;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import nl.sogeti.android.gpstracker.ng.util.FragmentTestRule;
import nl.sogeti.android.gpstracker.ng.utils.TrackUriExtensionKt;


@RunWith(AndroidJUnit4.class)
public class TrackListFragmentTest {

    @Rule
    public FragmentTestRule<TrackListFragment> wrapperFragment = new FragmentTestRule<>(TrackListFragment.class, false, false);
    private TrackListFragment sut = null;

    @Before
    public void setup() {
        sut = wrapperFragment.getFragment();
    }

    @Test
    public void testResumeStartsPresenter() {
        // Execute
        wrapperFragment.launchFragment(null);

        // Verify
        Assert.assertNotNull(sut.getTracksPresenter().getContext());
    }

    @Test
    public void testPauseStopsPresenter() {
        // Execute
        wrapperFragment.launchFragment(null);
        wrapperFragment.finishFragment();

        // Verify
        Assert.assertNull(sut.getTracksPresenter().getContext());
    }

    @Test
    public void testTrackSelectionInvokesListener() {
        // Prepare
        final Uri[] uries = new Uri[1];
        Uri uri = TrackUriExtensionKt.trackUri(2);
        TrackListFragment.Listener myListener = new TrackListFragment.Listener() {
            @Override
            public void onTrackSelected(@NotNull Uri uri) {
                uries[0] = uri;
            }
        };
        sut.setListener(myListener);

        // Execute
        sut.onTrackSelected(uri);

        // Verify
        Assert.assertEquals(uries[0], uri);
    }
}

File: studio/app/src/androidTest/java/nl/sogeti/android/gpstracker/ng/tracks/TrackListFragmentTest.kt
//package nl.sogeti.android.gpstracker.ng.tracks
//
//import android.content.ContentUris
//import android.support.test.rule.ActivityTestRule
//import android.support.test.runner.AndroidJUnit4
//import android.support.v7.app.AppCompatActivity
//import junit.framework.Assert
//import nl.sogeti.android.gpstracker.integration.ContentConstants.Tracks.TRACKS_URI
//import org.junit.Before
//import org.junit.Rule
//import org.junit.Test
//import org.junit.runner.RunWith
//import org.mockito.Mockito.mock
//import org.mockito.Mockito.verify
//import org.mockito.MockitoAnnotations
//
//@RunWith(AndroidJUnit4::class)
//class TrackListFragmentTest : AppCompatActivity() {
//
//    @Rule
//    public val wrapperActivity = ActivityTestRule<TrackListFragmentTest>(TrackListFragmentTest::class.java)
//    private var sut: TrackListFragment? = null
//
//    @Before
//    fun setup() {
//        MockitoAnnotations.initMocks(this)
//        sut = TrackListFragment()
//        wrapperActivity.activity.supportFragmentManager.beginTransaction().add(sut, "SUT").commitNow()
//    }
//
//    @Test
//    fun testResumeStartsPresenter() {
//        // Prepare
//        val sut = this.sut!!
//
//        // Execute
//        sut.onResume()
//
//        // Verify
//        Assert.assertNotNull(sut.tracksPresenter.context)
//    }
//
//    @Test
//    fun testPauseStopsPresenter() {
//        // Prepare
//        val sut = this.sut!!
//
//        // Execute
//        sut.onResume()
//
//        // Verify
//        Assert.assertNull(sut.tracksPresenter.context)
//    }
//
//    @Test
//    fun testTrackSelectionInvokesListener() {
//        // Prepare
//        val sut = this.sut!!
//        val mockListener = mock(TrackListFragment.Listener::class.java)
//        val uri = ContentUris.withAppendedId(TRACKS_URI, 2)
//        sut.listener = mockListener
//
//        // Execute
//        sut.onTrackSelected(uri)
//
//        // Verify
//        verify(mockListener).onTrackSelected(uri)
//    }
//}